mining.go

```diff
@@ -2076,6 +2077,7 @@ func newBgBlkTmplGenerator(tg *BlkTmplGenerator, addrs []dcrutil.Address, permit
 		subscribers:                make(map[chan *BlockTemplate]struct{}),
 		templatePool:               make(map[[templateKeySize]byte]*wire.MsgBlock),
 		permitConnectionlessMining: permitConnectionlessMining,
+		ctx:                        ctx,
```

Comment:
Please revert all of these extra `ctx` changes.  It goes against the spirit of how it's supposed to work which is that the context is passed into the functions which have the scope in question.

Reply:
For reference, see the context documentation which specifically says:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx: